### DISCUSSION TITLE
How can product type be created in Medusa?

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4847] [Q&A]
I am aware that in Medusa, there is a `product_type` table and that can be used to differentiate product, based on the discussion here: https://github.com/medusajs/medusa/discussions/4478

But when I check the API documentation, there is no endpoint for creating new product type. For the [product type service](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/services/product-type.ts), there is no method for creating product type.

So what is the expected way to create a new product type in Medusa?

**Comment by genox** [comment]
I would like to know the same. :)

**Comment by olivermrbl** [comment]
We currently don't have an endpoint for creating product types nor do we have a user interface for it in Admin. You'll have to add them as part of creating/updating a product. 

In Medusa Admin, they can be added in the "Organize" section:
<img width="784" alt="CleanShot 2023-09-21 at 12 56 24@2x" src="https://github.com/medusajs/medusa/assets/59018053/76b4a5eb-a7c7-49e7-bfad-26719600f24d">

After a product type has been added as part of creating a product, you'll be able to reuse it for subsequent product creations. 

In terms of raw payload, these two options are differentiated by the latter requiring an id in the payload. See the examples below:

Creating a new type as part of creating a product:
```
// POST /admin/products
{
   ...
   type: { value: "some type" }
}
```

Using an existing type as part of creating a product:
```
// POST /admin/products
{
   ...
   type: { value: "some type", id: "some_type_id" }
}
```

**Comment by genox** [comment]
Or, if you run pgadmin alongside your stack, it might be easier to add a new row to product_types directly. Well, I assume this should work, I will test in on my dev setup.
